packages/menu/src/MenuItem.ts

```diff
@@ -211,24 +172,15 @@ export class MenuItem extends LikeAnchor(
         if (this._itemChildren) {
             return this._itemChildren;
         }
-
-        const iconSlot = this.shadowRoot?.querySelector(
-            'slot[name="icon"]'
-        ) as HTMLSlotElement;
-        const icon = !iconSlot
-            ? []
-            : iconSlot.assignedElements().map((element) => {
-                  const newElement = element.cloneNode(true) as HTMLElement;
-                  newElement.removeAttribute('slot');
-                  newElement.classList.toggle('icon');
-                  return newElement;
-              });
-        const contentSlot = this.shadowRoot?.querySelector(
-            'slot:not([name])'
-        ) as HTMLSlotElement;
-        const content = !contentSlot
-            ? []
-            : contentSlot.assignedNodes().map((node) => node.cloneNode(true));
+        const icon = this.iconSlot.assignedElements().map((element) => {
+            const newElement = element.cloneNode(true) as HTMLElement;
+            newElement.removeAttribute('slot');
+            newElement.classList.toggle('icon');
+            return newElement;
+        });
+        const content = this.contentSlot
+            .assignedNodes()
+            .map((node) => node.cloneNode(true));
```

Comment:
Isn't it possible for `contentSlot` or `iconSlot` to be undefined / null?

Reply:
In the frame before the first call to `render()`, sure there is a moment in which these are not yet defined. There's likely harder to define issues that arise in that moment than this, but we'd love to hear if you happen to have a use case where that arises! Then we can make sure it's covered in our unit test suite and resilient to any subsequent changes that may come to this part of the library.

Reply:
Thanks! I created #3626.